raysect/core/math/function/function1d/base.pyx

```diff
@@ -62,82 +62,82 @@ cdef class Function1D:
         return self.evaluate(x)
 
     def __add__(object a, object b):
-        if isinstance(a, Function1D):
-            if isinstance(b, Function1D):
+        if is_callable(a):
+            if is_callable(b) or callable(b):
```

Comment:
Isn't the `or callable (b)` here redundant? This check is already performed in `is_callable(b)`.

Reply:
Thanks, that was a find-replace mistake.